src/main/remote-sync/RemoteSyncManager.ts

```diff
@@ -0,0 +1,347 @@
+          : 'INITIAL_FILES_SYNC',
+      });
+      if (syncConfig.retry >= syncConfig.maxRetries) {
+        // No more retries allowed,
+        this.filesSyncStatus = 'SYNC_FAILED';
+        this.checkRemoteSyncStatus();
+        return;
+      }
+
+      await this.syncRemoteFiles({
+        retry: syncConfig.retry + 1,
+        maxRetries: syncConfig.maxRetries,
+      });
+    }
+  }
+
+  /**
+   * Syncs all the remote folders and saves them into the local db
+   * @param syncConfig Config to execute the sync with
+   * @returns
+   */
+  private async syncRemoteFolders(syncConfig: SyncConfig) {
+    const lastFoldersSyncAt = await helpers.getLastFoldersSyncAt();
+    try {
+      const { hasMore, result } = await this.fetchFoldersFromRemote(
+        lastFoldersSyncAt
+      );
+
+      for (const remoteFolder of result) {
+        // eslint-disable-next-line no-await-in-loop
+        await this.createOrUpdateSyncedFolderEntry(remoteFolder);
+        const foldersUpdatedAt = new Date(remoteFolder.updatedAt);
+        if (
+          !lastFoldersSyncAt ||
+          helpers.lastSyncedAtIsNewer(
+            foldersUpdatedAt,
+            lastFoldersSyncAt,
+            SYNC_OFFSET_MS
+          )
+        ) {
+          Logger.info(`Saving folders updatedAt ${foldersUpdatedAt}`);
+          helpers.saveLastFoldersSyncAt(foldersUpdatedAt, SYNC_OFFSET_MS);
+        }
+      }
+
+      if (!hasMore) {
+        this.foldersSyncStatus = 'SYNCED';
+        this.checkRemoteSyncStatus();
+        return;
+      }
+
+      Logger.info('Retrieving more folders for sync');
+      await this.syncRemoteFolders({
+        retry: 1,
+        maxRetries: syncConfig.maxRetries,
+      });
+    } catch (error) {
+      Logger.error('Remote folders sync failed ', error);
+      reportError(error as Error, {
+        lastFoldersSyncAt: lastFoldersSyncAt
+          ? lastFoldersSyncAt.toISOString()
+          : 'INITIAL_FOLDERS_SYNC',
+      });
+      if (syncConfig.retry >= syncConfig.maxRetries) {
+        // No more retries allowed,
+        this.foldersSyncStatus = 'SYNC_FAILED';
+        this.checkRemoteSyncStatus();
+        return;
+      }
+
+      await this.syncRemoteFolders({
+        retry: syncConfig.retry + 1,
+        maxRetries: syncConfig.maxRetries,
+      });
+    }
+  }
+
+  /**
+   * Fetch the files that were updated after the given date
+   *
+   * @param updatedAtCheckpoint Retrieve files that were updated after this date
+   */
+  private async fetchFilesFromRemote(updatedAtCheckpoint?: Date): Promise<{
+    hasMore: boolean;
+    result: RemoteSyncedFile[];
+  }> {
+    const token = obtainToken('newToken');
+
+    let query = `?limit=${this.config.fetchFilesLimitPerRequest}&offset=0&status=ALL`;
+
+    if (updatedAtCheckpoint) {
+      query += `&updatedAt=${updatedAtCheckpoint.toISOString()}`;
+    }
+
+    const path = `${process.env.NEW_DRIVE_URL}/drive/files${query}`;
+
+    const response = await fetch(path, {
```

Comment:
We added an Axios client with the authentication built-in and with unauthorized response handler. It can be obtained by the `getClient` function on `src/shared/HttpClient/main-process-client.ts`
 [src/shared/HttpClient/HttpClient.ts](https://github.com/internxt/drive-desktop/blob/8cd7c83f18f7ac064a10fffb440f9e67453df9b7/src/shared/HttpClient/HttpClient.ts#L12)

Reply:
I added an httpClient that we pass via constructor